### PR TITLE
Upgrade kubernetes-client-api to 6.8.1

### PR DESCRIPTION
**Pull request by Vlatombe**
Downstream of https://github.com/jenkinsci/kubernetes-client-api-plugin/pull/221

Breaking changes (from library):
* Migration to SnakeYAML engine (regression in octal notation support)

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by julieheard** [comment]
I notice that this plugin has a dependency on kubernetes-credentials-plugin. Do we need to make sure https://github.com/jenkinsci/kubernetes-credentials-plugin/pull/39 is merged and released first, so we can use the release version for that in this plugin?

**Comment by Vlatombe** [comment]
It's probably safer.

File: pom.xml [hunk 1]
```diff
@@ -79,7 +81,8 @@
     <dependency>
       <groupId>org.jenkinsci.plugins</groupId>
       <artifactId>kubernetes-credentials</artifactId>
-      <version>0.10.0</version>
+      <!-- TODO https://github.com/jenkinsci/kubernetes-credentials-plugin/pull/39 @ f62dc9ee13617d84ec899274ed897668a4703242 -->
+      <version>0.10.1-rc150.f62dc9ee1361</version>
```
Comment (Vlatombe) [inline pom.xml:85]: https://github.com/jenkinsci/kubernetes-credentials-plugin/pull/39

File: pom.xml [hunk 2]
```diff
@@ -264,6 +267,12 @@
         <artifactId>mockwebserver</artifactId>
         <version>4.9.3</version><!-- should be aligned with the version provided by okhttp-api-plugin -->
       </dependency>
+      <dependency>
+        <groupId>io.jenkins.plugins</groupId>
+        <artifactId>snakeyaml-api</artifactId>
+        <!-- TODO https://github.com/jenkinsci/snakeyaml-api-plugin/pull/94 @ 4960f40749788fcfe6ddc7def8e0bbc0ce937046-->
+        <version>2.1-110.v4960f4074978</version>
```
Comment (Vlatombe) [inline pom.xml:274]: https://github.com/jenkinsci/snakeyaml-api-plugin/pull/94

File: pom.xml [hunk 3]
```diff
@@ -58,7 +60,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>kubernetes-client-api</artifactId>
-      <version>6.4.1-215.v2ed17097a_8e9</version>
+      <version>6.8.1-220.v1f66736f9a_42</version>
```
Comment (Vlatombe) [inline pom.xml:63]: https://github.com/jenkinsci/kubernetes-client-api-plugin/pull/221

**Comment by Vlatombe** [comment]
@jglick Any clue about this new warning raised in `org.csanchez.jenkins.plugins.kubernetes.pipeline.KubernetesPipelineTest.runInPod` ?

```
   7.725 [id=1175]	WARNING	o.j.p.w.flow.FlowExecutionList#unregister: Owner[run In Pod/1:run In Pod #1] was not in the list to begin with: []
```

**Comment by Vlatombe** [comment]
Looks like it never registers in the first place.

**Comment by Vlatombe** [comment]
https://github.com/jenkinsci/workflow-api-plugin/pull/304 seems to fix it

**Comment by jglick** [comment]
Judging by the stack traces, it looks like a race condition: depending on timing, the `AgentReconnectionListener` may or may not call `StepExecution.applyAll` after the `FlowExecution.complete` but before `WorkflowRun.finish` gets to that point. Indeed we do not need to unregister anything from `StepExecution.applyAll`.

File: pom.xml [hunk 4]
```diff
@@ -58,7 +60,7 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>kubernetes-client-api</artifactId>
-      <version>6.4.1-215.v2ed17097a_8e9</version>
+      <version>6.8.1-223.vefe792551b_b_d</version>
```
Comment (Vlatombe) [inline pom.xml:63]:
```suggestion
      <version>6.8.1-224.vd388fca_4db_3b_</version>
```

File: pom.xml [hunk 5]
```diff
@@ -79,12 +81,13 @@
     <dependency>
       <groupId>org.jenkinsci.plugins</groupId>
       <artifactId>kubernetes-credentials</artifactId>
-      <version>0.10.0</version>
+      <!-- TODO https://github.com/jenkinsci/kubernetes-credentials-plugin/pull/39 @ 837cce3529e6f8d6fb0e4ae9b27eb6666cf0d3ca -->
+      <version>0.10.1-rc152.837cce3529e6</version>
```
Comment (Vlatombe) [inline pom.xml:85]:
```suggestion
      <version>0.11</version>
```

**Review by julieheard — APPROVED** [review]

**Comment by amuniz** [comment]
As part of the snakeyaml (transitive) update a breaking change was introduced. The new `snakeyaml-engine` is implementing YAML 1.2.2, in which the merge operator (`<<`) is not defined anymore (it was removed from the spec). So any yaml using it will fail to parse.

I think it's worth noting in release notes @Vlatombe?

**Comment by Vlatombe** [comment]
Do you have a concrete use case for it ? I've never seen this operator being used in the context of this plugin

**Comment by Vlatombe** [comment]
Probably a general mention that snakeyaml-engine implements a new version of the yaml spec, and so anything that was relying on specifics of yaml 1.1 that got changed or removed in yaml 1.2 no longer works.

**Comment by Vlatombe** [comment]
Updated https://github.com/jenkinsci/kubernetes-plugin/releases/tag/4029.v5712230ccb_f8

**Comment by vicecea** [comment]
Facing the same issues as https://issues.jenkins.io/browse/JENKINS-71956
Any chance we can get an additional option that is backwards compatible?

Would there be a way to fix this in `snakeyaml-engine` ?

**Pull request merged**